### PR TITLE
Fix decimal parsing of Hive data

### LINKED ISSUE #7250: Error parsing decimal using RCFile Text
With
```
set session hive.rcfile_optimized_reader_enabled = false;
```

following may occur

```
java.lang.ArithmeticException: Rounding necessary
java.math.BigDecimal.commonNeedIncrement(BigDecimal.java:4148)
java.math.BigDecimal.needIncrement(BigDecimal.java:4204)
java.math.BigDecimal.divideAndRound(BigDecimal.java:4112)
java.math.BigDecimal.setScale(BigDecimal.java:2452)
java.math.BigDecimal.setScale(BigDecimal.java:2386)
com.facebook.presto.spi.type.Decimals.rescale(Decimals.java:259)
com.facebook.presto.hive.ColumnarTextHiveRecordCursor.parseDecimalColumn(ColumnarTextHiveRecordCursor.java:507)
com.facebook.presto.hive.ColumnarTextHiveRecordCursor.parseDecimalColumn(ColumnarTextHiveRecordCursor.java:495)
com.facebook.presto.hive.ColumnarTextHiveRecordCursor.parseColumn(ColumnarTextHiveRecordCursor.java:625)
com.facebook.presto.hive.ColumnarTextHiveRecordCursor.isNull(ColumnarTextHiveRecordCursor.java:583)
com.facebook.presto.hive.HiveCoercionRecordCursor.isNull(HiveCoercionRecordCursor.java:150)
com.facebook.presto.hive.HiveRecordCursor.isNull(HiveRecordCursor.java:237)
com.facebook.presto.spi.RecordPageSource.getNextPage(RecordPageSource.java:107)
com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:256)
com.facebook.presto.operator.Driver.processInternal(Driver.java:378)
com.facebook.presto.operator.Driver.processFor(Driver.java:301)
com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:622)
com.facebook.presto.execution.TaskExecutor$PrioritizedSplitRunner.process(TaskExecutor.java:555)
com.facebook.presto.execution.TaskExecutor$Runner.run(TaskExecutor.java:691)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
java.lang.Thread.run(Thread.java:745)

### PR DESCRIPTION
**Pull request by findepi**
Fixes #7250, _assuming_ the expected behavior is "do like Hive does (round)" rather than "throw a more meaningful exception".

### CODE REVIEW

**Comment by findepi** [comment]
With ```set session hive.rcfile_optimized_reader_enabled = true;``` a different reader is used, different code path and a different exception raised. I will check tomorrow.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDecimalParser.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        checkParseDecimal("-3.25", 2, 1, new BigDecimal("-3.3"));
+    }
+
+    private void checkParseDecimal(String input, int precision, int scale, BigDecimal expected)
+    {
+        byte[] bytes = input.getBytes(US_ASCII);
+        BigDecimal parsed = HiveDecimalParser.parseHiveDecimal(bytes, 0, bytes.length, DecimalType.createDecimalType(precision, scale));
+        assertEquals(parsed, expected);
+    }
+}
```
Comment (losipiuk) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDecimalParser.java]: new line

Comment (losipiuk) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDecimalParser.java]: upate your IntelliJ config

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDecimalParser.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static org.testng.Assert.*;
+
+public class TestHiveDecimalParser
+{
+    @Test
+    public void testParseDecimal()
```
Comment (losipiuk) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDecimalParser.java]: does one of this tests fail before introducing fix in next commit?

Comment (findepi) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDecimalParser.java]: Yes, the ones involving rounding, where parsed string is longer than the string used to construct expected BD value.

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**